code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalExtension.kt

```diff
@@ -0,0 +1,148 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.signal
+
+import com.adobe.marketing.mobile.*
+import com.adobe.marketing.mobile.services.HitQueuing
+import com.adobe.marketing.mobile.services.PersistentHitQueue
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.utils.DataReader
+
+class SignalExtension(extensionApi: ExtensionApi?) : Extension(extensionApi) {
+    private val hitQueue: HitQueuing
+
+    companion object {
+        private const val LOG_TAG = "SignalExtension"
+    }
+
+    init {
+        val dataQueue =
+            ServiceProvider.getInstance().dataQueueService.getDataQueue(SignalConstants.EXTENSION_NAME)
+        hitQueue = PersistentHitQueue(dataQueue, SignalHitProcessor())
+    }
+
+    override fun onRegistered() {
+        api?.registerEventListener(EventType.RULES_ENGINE, EventSource.RESPONSE_CONTENT) {
+            handleRulesEngineResponse(it)
+        }
+        api?.registerEventListener(EventType.CONFIGURATION, EventSource.RESPONSE_CONTENT) {
+            handleConfigurationResponse(it)
+        }
+    }
+
+    override fun getName(): String {
+        return SignalConstants.EXTENSION_NAME
+    }
+
+    override fun getFriendlyName(): String {
+        return SignalConstants.FRIENDLY_NAME
+    }
+
+    override fun getVersion(): String {
+        return Signal.extensionVersion()
+    }
+
+    private fun handleConfigurationResponse(event: Event?) {
```

Comment:
I added `Nonnull` annotation to `ExtensionEventListener`. We can update all eventListeners to `Event` instead of `Event?`

Reply:
Will do.